metricbeat/mb/module/connector.go

```diff
@@ -36,20 +38,25 @@ type Connector struct {
 
 type connectorConfig struct {
 	Processors processors.PluginConfig `config:"processors"`
+	// ES output index pattern
+	Index fmtstr.EventFormatString `config:"index"`
 
 	// KeepNull determines whether published events will keep null values or omit them.
 	KeepNull bool `config:"keep_null"`
 
 	common.EventMetadata `config:",inline"` // Fields and tags to add to events.
 }
 
-func NewConnector(pipeline beat.Pipeline, c *common.Config, dynFields *common.MapStrPointer) (*Connector, error) {
+func NewConnector(
+	beatInfo beat.Info, pipeline beat.Pipeline,
```

Comment:
I've been torn about this sort of change in one of my PRs as well. Basically, we're now passing an extra `beat.Info` parameter to this function. Looking at the calls to this constructor, they now look like this:

```
module.NewConnector(b.Info, b.Publisher, ...)
```

where `b` is of type `*beat.Beat`. 

I wonder if should just make this constructor take `b *beat.Beat` instead of `beatInfo beat.Info, pipeline beat.Pipeline`. I could go either way on this. WDYT?

Reply:
My usual preference is to pass the simpler information, both to reduce contextual load (a `BeatInfo` parameter implies a lot more about the function's behavior than a broadly-scoped container that could be affecting many things in non-obvious ways) and because it's easy to refactor to `beat.Beat` in the future if we ever really need to, but hard to take it back once it's in the API and people start depending on that field in unexpected ways :-)

Reply:
Yup, that's fair. I think my concern of adding more arguments is not as strong in comparison, at least not at this point, where the function would be taking in only 4 arguments. If that number ever grows to be really unwieldy, we can revist.

+1 for keeping this change as-is.